File: Week 7/my-backend/seed.js
const mongoose = require('mongoose');
const Product = require('./models/Product'); // Nhúng Model của bạn

// Danh sách dữ liệu mẫu bạn muốn thêm vào
const sampleProducts = [
  { name: "Laptop Dell XPS 15", price: 1500, category: "Electronics" },
  { name: "MacBook Pro M2", price: 2000, category: "Electronics" },
  { name: "Bàn phím cơ Keychron K8", price: 100, category: "Accessories" },
  { name: "Chuột Logitech MX Master 3S", price: 120, category: "Accessories" },
  { name: "Màn hình LG UltraFine 4K", price: 600, category: "Electronics" },
  { name: "Balo chống nước The North Face", price: 50, category: "Fashion" },
  { name: "Tai nghe Sony WH-1000XM5", price: 350, category: "Electronics" }
];

// Mở kết nối tới Database (Đổi thành localhost nếu máy bạn đang dùng localhost)
mongoose.connect('mongodb://localhost:27017/product_management')
  .then(async () => {
    console.log('⏳ Đang kết nối MongoDB để tạo dữ liệu mẫu...');

    // (Tùy chọn) Xóa sạch dữ liệu cũ để tránh bị trùng lặp khi chạy nhiều lần
    await Product.deleteMany({});
    console.log('🧹 Đã dọn dẹp dữ liệu cũ!');

    // Bơm toàn bộ mảng dữ liệu mẫu vào DB
    await Product.insertMany(sampleProducts);
    console.log(`🎉 Đã thêm thành công ${sampleProducts.length} sản phẩm mẫu!`);

    // Thêm xong thì đóng kết nối lại
    mongoose.connection.close();
  })
  .catch(err => {
    console.error('❌ Lỗi:', err);
    mongoose.connection.close();
  });

File: Week 7/python/flask-pymongo-api/seed.py
from pymongo import MongoClient
import sys

# Khởi tạo dữ liệu mồi (Mock Data)
SEED_PRODUCTS = [
    {
        "name": "Laptop Dell XPS 15",
        "price": 1800,
        "category": "Electronics"
    },
    {
        "name": "Bàn phím cơ Keychron K8",
        "price": 85,
        "category": "Accessories"
    },
    {
        "name": "Sách 'Clean Code'",
        "price": 35,
        "category": "Books"
    },
    {
        "name": "Màn hình LG UltraWide 34 inch",
        "price": 450,
        "category": "Electronics"
    },
    {
        "name": "Balo chống nước Mark Ryden",
        "price": 40,
        "category": "Fashion"
    }
]

def run_seed():
    try:
        # 1. Kết nối tới MongoDB
        client = MongoClient("mongodb://localhost:27017/", serverSelectionTimeoutMS=5000)
        db = client["product_management_db"]
        collection = db["products"]

        print("Đang kết nối MongoDB...")
        
        # 2. Xóa dữ liệu cũ (để tránh bị trùng lặp nếu bạn chạy file này nhiều lần)
        collection.delete_many({})
        print("🗑️ Đã xóa sạch dữ liệu cũ trong collection 'products'.")

        # 3. Thêm dữ liệu mới
        result = collection.insert_many(SEED_PRODUCTS)
        
        print(f"✅ Seeding thành công! Đã thêm {len(result.inserted_ids)} sản phẩm vào Database.")
        
    except Exception as e:
        print(f"❌ Lỗi khi chạy seed data: {e}")
        sys.exit(1)

if __name__ == "__main__":
    run_seed()